src/subcommand/server.rs

```diff
@@ -832,6 +837,75 @@ impl Server {
     })
   }
 
+  async fn inscription_details(
+    Extension(server_config): Extension<Arc<ServerConfig>>,
+    Extension(index): Extension<Arc<Index>>,
+    Path(inscription_id): Path<InscriptionId>,
+  ) -> ServerResult<Response> {
+    if !index.has_sat_index() {
+      return Err(ServerError::NotFound(
+        "this server has no sat index".to_string(),
+      ));
+    }
```

Comment:
I think the JSON should just have the sat field as an option instead of requiring a sat index for this endpoint.